### DISCUSSION TITLE
Percentage discount with specific product option added doesn't show the amount of the discount.

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#11817] [Q&A] [answered]
Hi, there's another problem with the PERCENTAGE DISCOUNT when you select SPECIFIC PRODUCT option.

Is similar to another issue I found in the past https://github.com/saleor/saleor/discussions/10528.
 If you don't apply the specific product condition the discount works as expected.

If you add the specific product when you apply the percentage discount the amount showed is 0.

Below the example:

- I create my percentage discount with specific product assigned (it doesn't works even if you assign or not a category or collection or product so I show both cases)

**with category assigned**
<img width="992" alt="Screenshot 2023-01-26 at 09 27 50" src="https://user-images.githubusercontent.com/38940340/214790077-a47c5f57-5393-49fc-869e-71624f2ec369.png">

**with any category or collection or product assigned**
<img width="828" alt="Screenshot 2023-01-26 at 09 27 56" src="https://user-images.githubusercontent.com/38940340/214790080-0b37a6cd-4965-4f07-995f-7a4e94b291ee.png">

- I use the code in the checkout in the `checkoutAddPromoCode` query:
```
{
    "variables": {
        "id": "XXXXXXXXXXXXXXXXX",
        "promoCode": "EXPO25",
        "languageCode": "IT"
    }
}
```

- the response has as the amount 0.
```

{
    "checkoutAddPromoCode": {
        "checkout": {
            "discount": {
                "currency": "EUR",
                "amount": 0
            },
            "giftCards": [],
            "discountName": null,
            "translatedDiscountName": "",
            "voucherCode": "EXPO25",
        },
        "errors": []
    }
}
```

I'm asking you if there will be a fix (I hope not so far from today :( ) because as the previous issue it create a confusing experience and we just release two new ecommerce with Saleor so the promo code is one important tool to grow up the sells.

Really thanks in advance and kind regards,

Davide

**Comment by maarcingebala** [comment, marked as answer]
Hey, for `SPECIFIC_PRODUCT` voucher the discount amount is not set, instead the discount is distributed between checkout/order lines. Vouchers for entire order are calculated into this `discount.amount` field.

**Reply by DidoMarchet** [reply]
Hi @maarcingebala sonit's not a bug sorry I update the title. Now I ask they check the answer so if ok I let you know.
Thanks in advance and kind regards,
Davide
